### PR TITLE
Expand the annotation service to support GO annotation

### PR DESCRIPTION
**Pull request by leungmanhin**
As suggested in https://github.com/MOZI-AI/annotation-scheme/issues/192.
The changes made in this PR is most probably incomplete to fully annotate GO terms, comments would be much appreciated

### CODE REVIEW

File: annotation/functions.scm [hunk 1]
```diff
@@ -885,3 +885,16 @@
...
     (cog-incoming-by-type go-term 'MemberLink))
 )
+
+(define (find-go-protein go-term)
+"
+  find-go-protein GO-TERM
+
+  Find the proteins associate with GO-TERM via a MemberLink.
+"
+  (filter
```
Comment (Habush) [inline annotation/functions.scm:913]: I think using the following pattern matcher query would be more efficient

```
(Get 
    (TypedVariable (Variable "$prot") (TypedNode "MoleculeNode"))
    (Member 
       (Variable "$prot")
       go-term)
) 
```

Comment (leungmanhin) [reply annotation/functions.scm:930]: Thanks, I've just changed it

File: annotation/functions.scm [hunk 2]
```diff
@@ -898,3 +898,17 @@
            (equal? (cog-type (gar memblink)) 'MoleculeNode)))
     (cog-incoming-by-type go-term 'MemberLink))
 )
+
+(define (find-go-parents go-term)
```
[the remark this answers is not in the document]
Comment (leungmanhin) [reply annotation/functions.scm]: The `find-parent` function you pointed out returns really only the node representing the parent of the input GO term, e.g. `(ConceptNode "GO:0042660")`, but I guess the service expects the whole `InheritanceLink`?

File: annotation/functions.scm [hunk 3]
```diff
@@ -912,3 +912,23 @@
            (string-prefix? "GO:" (cog-name (gdr inhlink)))))
     (cog-incoming-by-type go-term 'InheritanceLink))
 )
+
+(define (find-go-namespace go-term)
```
[the remark this answers is not in the document]
Comment (leungmanhin) [reply annotation/functions.scm]: The `find-GO-ns` function you pointed out returns really only the node representing the namespace of the input GO term, e.g. `(ConceptNode "molecular_function")`, but I guess the service expects the whole `EvaluationLink`?

File: annotation/functions.scm [hunk 4]
```diff
@@ -932,3 +932,23 @@
         (Predicate "GO_namespace")
         (List go-term var-ns))))
 )
+
+(define (find-go-name go-term)
```
[the remark this answers is not in the document]
Comment (leungmanhin) [reply annotation/functions.scm]: Yes I'm already using this function

File: annotation/functions.scm [hunk 5]
```diff
@@ -972,3 +972,46 @@
         (Predicate "GO_has_part")
         (List go-term var-go-term))))
 )
+
+(define (find-go-regulates go-term)
```
[the remark this answers is not in the document]
Comment (leungmanhin) [reply annotation/functions.scm]: Sure I'm fine either way

[the remark this answers is not in the document]
Comment (Habush) [reply annotation/functions.scm]: Please checkout #212

**Pull request merged**